app/models/katello/subscription_status.rb

```diff
@@ -23,6 +23,8 @@ def to_label(_options = {})
     end
 
     def to_global(_options = {})
+      return ::HostStatus::Global::WARN if host.subscription_facet.hypervisor && host.subscription_facet.candlepin_consumer.entitlements.empty?
```

Comment:
this would only effect the global status, but i think we want the subscription status to reflect this as well, so i think you want this logic in the to_status() method.  In addition I think this should be its own state, so that it can be searched for specifically (similar to the UNKNOWN status).